silkrpc/types/block.hpp

```diff
@@ -35,6 +37,70 @@ struct Block : public silkworm::BlockWithHash {
 
 std::ostream& operator<<(std::ostream& out, const Block& b);
 
+class BlockNumberOrHash {
+public:
+    BlockNumberOrHash() {}
```

Comment:
You could remove default constructor and undefined state, that in my understanding are present just for deserialization from `nlohmann::json` objects, by applying a specific technique outlined [here](https://github.com/nlohmann/json#how-can-i-use-get-for-non-default-constructiblenon-copyable-types)

Reply:
Done